agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -59,116 +62,133 @@ public NetworkStatsbeat(CustomDimensions customDimensions) {
 
   @Override
   protected void send(TelemetryClient telemetryClient) {
-    Map<String, IntervalMetrics> local;
+    Map<IntervalMetricsKey, IntervalMetrics> local;
     synchronized (lock) {
       local = new HashMap<>(instrumentationKeyCounterMap);
       instrumentationKeyCounterMap.clear();
     }
 
-    for (Map.Entry<String, IntervalMetrics> entry : local.entrySet()) {
-      String ikey = entry.getKey();
-      sendIntervalMetric(telemetryClient, ikey, entry.getValue());
+    for (Map.Entry<IntervalMetricsKey, IntervalMetrics> entry : local.entrySet()) {
+      IntervalMetricsKey key = entry.getKey();
+      sendIntervalMetric(telemetryClient, key, entry.getValue());
     }
   }
 
   public void incrementRequestSuccessCount(long duration, String ikey, String host) {
     doWithIntervalMetrics(
         ikey,
         host,
+        null,
         intervalMetrics -> {
           intervalMetrics.requestSuccessCount.incrementAndGet();
           intervalMetrics.totalRequestDuration.getAndAdd(duration);
         });
   }
 
-  public void incrementRequestFailureCount(String ikey, String host) {
+  public void incrementRequestFailureCount(String ikey, String host, Integer statusCode) {
```

Comment:
I think it makes the method signature clearer since non-null, and let auto-boxing (or explicit boxing) happen later on
```suggestion
  public void incrementRequestFailureCount(String ikey, String host, int statusCode) {
```